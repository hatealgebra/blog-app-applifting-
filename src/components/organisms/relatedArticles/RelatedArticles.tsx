import React from "react";
import { components } from "../../../types/declarations";
import ArticlePreviewSmall from "../../molecules/articlePreview/ArticlePreviewSmall";
import { RelatedArticlesContainer } from "./relatedArticles.styled";

const RelatedArticles = ({ articles }: RelatedArticlesProps) => {
  return (
    <RelatedArticlesContainer className="related-articles">
      <h3>Related Articles</h3>
      <div className="related-articles__articles">
        {articles && articles.length !== 0
          ? articles.map((article) => (
              <ArticlePreviewSmall
                key={article.articleId}
                articleId={article.articleId!}
                heading={article.title!}
              >
                {article.perex!}
              </ArticlePreviewSmall>
            ))
          : "No articles to show"}
      </div>
    </RelatedArticlesContainer>
  );
};

interface RelatedArticlesProps {
  articles: components["schemas"]["Article"][] | null;
}

export default RelatedArticles;
