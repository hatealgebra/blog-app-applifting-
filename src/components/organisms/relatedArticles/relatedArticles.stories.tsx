import React from "react";
import RelatedArticles from "./RelatedArticles";
import relatedArticlesJSON from "../../../__mocks__/asyncData/get/allArticlesResponse.mock.json";

export const RelatedArticlesSection = () => (
  <RelatedArticles articles={relatedArticlesJSON.items} />
);

export default {
  title: "Organisms/Related Articles section",
  component: RelatedArticles,
};
