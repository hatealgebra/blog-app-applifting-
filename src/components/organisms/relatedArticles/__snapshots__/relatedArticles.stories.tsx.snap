// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Related Articles section RelatedArticlesSection smoke-test 1`] = `
<div class="sc-eflkNB cacjai related-articles">
  <h3>
    Related Articles
  </h3>
  <div class="related-articles__articles">
    <a class="sc-idyqAC iVyZKQ"
       href="/articles/c7b1f38a-e70f-4795-bdd8-802c68eb32dc"
    >
      <h5>
        British Cats: Charming Companions of the Isles
      </h5>
      <p>
        British cats, known for their distinctive charm and character, hold a special place in the hearts of cat enthusiasts around the wo...
      </p>
    </a>
    <a class="sc-idyqAC iVyZKQ"
       href="/articles/d1cfbb7d-e0d9-4341-a131-f1dd077a6fb0"
    >
      <h5>
        Bean the Cat: A Tale of Adventure, Friendship, and Whiskers
      </h5>
      <p>
        In the heart of a quaint little village nestled between rolling hills, a remarkable feline named Bean resided. Bean wasn't your ty...
      </p>
    </a>
    <a class="sc-idyqAC iVyZKQ"
       href="/articles/abd4dcae-d68c-4aa2-9651-e971d4389425"
    >
      <h5>
        The Irresistible Charm of Orange Cats: Exploring Their Purr-sonalities and Memes
      </h5>
      <p>
        Orange cats, with their vibrant coats and captivating personalities, have captured the hearts of cat lovers around the world. Thes...
      </p>
    </a>
    <a class="sc-idyqAC iVyZKQ"
       href="/articles/541d7c47-c7ec-4ec9-a2b0-b6a817278c40"
    >
      <h5>
        Unraveling the Mystery: The Enigmatic World of Grumpy Cats and Their Peculiar Demeanor
      </h5>
      <p>
        In the realm of feline companionship, there exist an intriguing subset of cats that have garnered attention not for their charming...
      </p>
    </a>
    <a class="sc-idyqAC iVyZKQ"
       href="/articles/2491852b-6566-4fae-ace3-4510cd40e81a"
    >
      <h5>
        The Fascinating World of Cat Sleeping Habits: Exploring Feline Slumber
      </h5>
      <p>
        Cats are renowned for their unique and often mysterious behaviors, and one of the most intriguing aspects of their lives is their ...
      </p>
    </a>
  </div>
</div>
`;
